### DISCUSSION TITLE
Export to CSV field separator

### DISCUSSION THREAD

**Discussion by lavoied** [sqlitebrowser/sqlitebrowser#2897] [Q&A]
I have a database containing so many different characters that it is hard to choose a field separator when exporting data to CSV.   Using the Export to CSV permit only 1 character maximum.   Is there a way to allow 2 characters (example @@)?

I also tried to write and SQL query to export the data to CSV with custom field separator, quote character and new line character but I wasn't able to figure it out.  is it possible to write such a query?

**Comment by chrisjlocke** [comment]
You can still use commas as a separator, but you just have to enclose the fields in quotes.  Is that possible?
So if your field contains "chris, is a muppet" (ie, a comma) you can still use commas as a separator ... just enclose it in spaces, so the line reads: `"chris, is a muppet","2000-01-01"`

Obviously this goes to pot if your field also contains quotation marks .. 

With regard to your query, you just need string concatenation.  So `select field1 || ',' || field2 || char(9) || field3 from table1`

char(9) is a tab character.  You can't use char(13) (new line) as DB4S probably strips these out to display correctly - it didn't like it when I tried.  But copying rows in the grid and pasting them obviously puts in a new line anyway.

But I'd try and stick with the comma as a separator if you can ... it's why it's called CSV after all.... ;)
